nucypher/blockchain/eth/clients.py

```diff
@@ -228,7 +233,7 @@ def check_for_timeout(t):
             last_update = maya.now()
             duration = (last_update - start_time).seconds
             if duration > t:
-                raise self.SyncTimeout
+                return True  # Just continue on.  # TODO: does this make sense?
```

Comment:
Eh? I don't understand why.
Neither do I understand sleep(0) below

Reply:
Okay - I restored the exception here.

Reply:
With regard to `time.sleep(0)` - Take a look at this stack overflow: <https://stackoverflow.com/questions/7273474/behavior-of-pythons-time-sleep0-under-linux-does-it-cause-a-context-switch>.  

A comment may be helpful here.